### DISCUSSION TITLE
LFX Mentorship 2023 03-Sep-Nov Challenge - for #2686

### DISCUSSION THREAD

**Discussion by q82419** [WasmEdge/WasmEdge#2697] [Show and tell]
# Introduction

For those who want to apply for LFX mentorship for #2686, this is a pretest for the application. The goal of this LFX mentorship is to support the AOT mode in proxy-wasm for WasmEdge runtime.

# Requirements

1. Build [proxy-wasm-cpp-host](https://github.com/proxy-wasm/proxy-wasm-cpp-host) with WasmEdge runtime in bazel.
2. Run the proxy-wasm-cpp-host tests for WasmEdge runtime.
3. Please snapshot your testing logs and upload here.

# Deadline

Please refer to [the timeline of LFX mentorship 2023 03-Sep-Nov](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/03-Sep-Nov).
Since we have to process the internal review and decide, the final date for valid submissions will be Aug 27, 5:00 PM PDT.

**Comment by vrnimje** [comment]
I have successfully built the proxy-wasm-cpp-host with WasmEdge in bazel, and also ran the tests
Here are the snapshots for the some of the testing logs. Is this ok, or should I attach snapshot all the logs?

![Screenshot from 2023-08-01 23-34-44](https://github.com/WasmEdge/WasmEdge/assets/103848930/d80f7a6e-96a0-48e5-abee-d6d2a6a69726)
![Screenshot from 2023-08-01 23-35-10](https://github.com/WasmEdge/WasmEdge/assets/103848930/286b8d23-f69c-4b84-9bc3-59bb845eced6)
![Screenshot from 2023-08-01 23-35-16](https://github.com/WasmEdge/WasmEdge/assets/103848930/92d687dc-f3f6-4cfe-a54a-b3ba79645402)
![Screenshot from 2023-08-01 23-35-35](https://github.com/WasmEdge/WasmEdge/assets/103848930/58f87e7b-ddae-400e-9f5f-3ada7ef87642)

**Reply by q82419** [reply]
It's fine. Thanks!

**Reply by vrnimje** [reply]
Hi @q82419 , I just came to know that my application for this mentorship was rejected. I would like to learn about why I wasn't chosen. I would really appreciate if you could provide me with some feedback regarding my application, it would really helpful for such future applications. Thanks!!

**Comment by shivaabhishek07** [comment]
Built successfully proxy-wasm-cpp-host with WasmEdge runtime in Bazel and ran tests.

![Screenshot 2023-08-06 165720](https://github.com/WasmEdge/WasmEdge/assets/107854894/ab533b8f-b72e-45df-bfc5-c0716ce7657e)
![Screenshot 2023-08-06 165735](https://github.com/WasmEdge/WasmEdge/assets/107854894/00e9003a-c546-48fc-8e0a-0e043356844c)
![Screenshot 2023-08-06 165805](https://github.com/WasmEdge/WasmEdge/assets/107854894/292a75c5-c6b8-4653-b897-2d9a99984cd6)
![Screenshot 2023-08-06 165819](https://github.com/WasmEdge/WasmEdge/assets/107854894/12ffe111-edb5-42a9-967d-516a148e2242)
![Screenshot 2023-08-06 165627](https://github.com/WasmEdge/WasmEdge/assets/107854894/89790484-73fa-49e8-839d-c5d2146a1d1d)
![Screenshot 2023-08-06 165644](https://github.com/WasmEdge/WasmEdge/assets/107854894/7be1d961-5e18-4d35-9053-113bfa2e0ccb)

**Comment by Sangam-ghimire** [comment]
Hey @q82419 I have completed the task. You can view the entire log at : https://github.com/Sangam-ghimire/proxy-wasm-cpp-host/blob/master/test_logs.txt.
I have attached few screenshot too below:
![image](https://github.com/WasmEdge/WasmEdge/assets/90495874/d2bcaa54-d73f-46be-804f-f54430f86c7e)
![image](https://github.com/WasmEdge/WasmEdge/assets/90495874/b9450637-7c6e-47c0-a142-c7fd7f27e8ac)
![image](https://github.com/WasmEdge/WasmEdge/assets/90495874/d7377297-ff1e-4808-bcab-7d86df171ebe)
![image](https://github.com/WasmEdge/WasmEdge/assets/90495874/e23b67a6-4987-4712-9423-06b73e46bd6c)
![image](https://github.com/WasmEdge/WasmEdge/assets/90495874/c4cc49a9-23bc-4c09-906a-e1a18e7047c5)
![image](https://github.com/WasmEdge/WasmEdge/assets/90495874/f8b88375-0be7-4c09-9a7d-10c0b9efd6d4)

**Comment by alabulei1** [comment]
Hi @Sangam-ghimire, @vrnimje, and @shivaabhishek07 

Thank you for completing the pretest. If you have not yet submitted your application for this project, please do so before 5 PM PDT on August 25th. 

Apply here: https://mentorship.lfx.linuxfoundation.org/project/3ee046ab-29d9-4e5c-bb7a-4564bd4e2765

**Comment by Shivkant-Chauhan** [comment]
Hey @q82419 @alabulei1 
I have completed the pre-requisite task of building the proxy-wasm-cpp-host with WasmEdge runtime in Bazel and running the tests. please refer to the screenshots below for the logs of the test runs --
(I am turning in the required tasks just before the deadlines, but this doesn't showcase my ability to work effectively and accomplish tasks ahead of schedule)

![Screenshot from 2023-08-27 17-11-28](https://github.com/WasmEdge/WasmEdge/assets/91013793/d2a7db4f-202b-4fe0-91bc-a713285519b7)
![Screenshot from 2023-08-27 17-11-42](https://github.com/WasmEdge/WasmEdge/assets/91013793/5002ea86-e711-4b55-964e-029ca21a0e86)
![Screenshot from 2023-08-27 17-11-49](https://github.com/WasmEdge/WasmEdge/assets/91013793/09f4163d-fed1-44d1-8aa3-0e4baa608ece)
![Screenshot from 2023-08-27 17-12-31](https://github.com/WasmEdge/WasmEdge/assets/91013793/a7be064c-18ca-421b-95b7-2bf67056ab6c)
![Screenshot from 2023-08-27 17-13-08](https://github.com/WasmEdge/WasmEdge/assets/91013793/6158d9e6-e65e-4dbb-9fa0-7af1824cca17)

**Comment by Shivkant-Chauhan** [comment]
@q82419 please let me know if there is anything else for me to do. thanks :)
